src/test/ui/lint/unused/unused-doc-comments-edge-cases.rs

```diff
@@ -26,4 +26,7 @@ fn doc_comment_on_expr(num: u8) -> bool {
     num == 3
 }
 
+fn doc_comment_on_generic<#[doc = "x"] T>(val: T) {}
+//~^ ERROR: unused doc comment
```

Comment:
Perhaps it's also worth adding tests for

```rust
fn doc_comment_on_generic<
    /// x
    T>(val: T) {}
//~^^ ERROR: unused doc comment
```

and

```rust
fn doc_comment_on_generic</** foo */ T>(val: T) {}
//~^ ERROR: unused doc comment
```

Reply:
Oops, GitHub didn't show me that this was already approved.

Reply:
I don't think we need to test those, anyway - rustc represents them both as `#[doc]` under the hood. There are plenty of other tests that would fail if that weren't the case.